XenAdmin/Controls/NetworkingTab/NetworkList.cs

```diff
@@ -732,31 +758,40 @@ private void launchVmNetworkSettingsDialog()
                 return;
 
             int device;
-            VIFDialog d;
-            if (int.TryParse(vif.device, out device))
-            {
-                d = new VIFDialog(vm.Connection, vif, device, vm.HasSriovRecommendation());
-            }
-            else
+            if (!int.TryParse(vif.device, out device))
             {
                 log.ErrorFormat("Aborting vif edit. Could not parse existing vif device to int. Value is: '{0}'", vif.device);
                 return;
             }
 
-            if (d.ShowDialog() != DialogResult.OK)
-                return;
+            VIF proxyVIF;
+            using (var d = new VIFDialog(vm.Connection, vif, device, vm.HasSriovRecommendation()))
+            {
+                if (d.ShowDialog() != DialogResult.OK)
+                    return;
+
+                proxyVIF = d.GetNewSettings();
+            }
 
-            var proxyVIF = d.GetNewSettings();
-            UpdateVIFCommand command = new UpdateVIFCommand(Program.MainWindow, vm, vif, proxyVIF);
+            var action = new UpdateVIFAction(vm, vif, proxyVIF);

```

Comment:
When a VIF is updated, the Events view shows 3 actions: updating, deleting and creating, because the update action runs the other two. We could hide the deletion and creation from the history. (this is existing behaviour, but we can improve it, since we look at this area)